apps/wallet/src/shared/deepBook/context.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useActiveAccount } from '_app/hooks/useActiveAccount';
+import { useCreateAccount, useMarketAccountCap } from '_hooks';
+import { useSuiClient } from '@mysten/dapp-kit';
+import { DeepBookClient } from '@mysten/deepbook';
+import { createContext, useContext, useEffect, useMemo, type ReactNode } from 'react';
+
+type DeepBookContextProps = {
+	client: DeepBookClient;
+};
+
+const DeepBookContext = createContext<DeepBookContextProps | null>(null);
+
+interface DeepBookContextProviderProps {
+	children: ReactNode;
+}
+
+export function useDeepBookContext() {
+	const context = useContext(DeepBookContext);
+	if (!context) {
+		throw new Error('useDeepBookContext must be used within a DeepBookContextProvider');
+	}
+	return context;
+}
+
+export function useDeepBookClient() {
+	return useDeepBookContext().client;
+}
+
+export function DeepBookContextProvider({ children }: DeepBookContextProviderProps) {
+	const suiClient = useSuiClient();
+	const activeAccount = useActiveAccount();
+
+	const { data, isLoading, refetch } = useMarketAccountCap(activeAccount?.address);
+
+	const accountCapId = data?.owner as string;
+
+	const deepBookClient = useMemo(() => {
+		return new DeepBookClient(suiClient, accountCapId);
+	}, [accountCapId, suiClient]);
+
+	const { mutate } = useCreateAccount({
+		onSuccess: refetch,
+		deepBookClient,
+	});
+
+	useEffect(() => {
+		if (!accountCapId && !isLoading) {
+			mutate();
```

Comment:
This sends a transaction without telling the user when this component mounts, which is both very scary, and not something we should do.  This also won't work for signers like ledger. We instead need to make this part of the explicit user flow.